python/ray/air/predictors/integrations/torch/torch_predictor.py

```diff
@@ -45,81 +46,54 @@ def from_checkpoint(
         model, preprocessor = load_checkpoint(checkpoint, model)
         return TorchPredictor(model=model, preprocessor=preprocessor)
 
-    # parity with Datset.to_torch
-    def _convert_to_tensor(
+    def _predict_pandas(
         self,
         data: pd.DataFrame,
-        feature_columns: Optional[
-            Union[List[str], List[List[str]], List[int], List[List[int]]]
-        ] = None,
-        dtypes: Optional[torch.dtype] = None,
-        unsqueeze: bool = True,
-    ) -> torch.Tensor:
-        """Handle conversion of data to tensor.
-
-        Same arguments as in ``convert_pandas_to_torch_tensor``."""
-        # TODO(amog): Add `_convert_numpy_to_torch_tensor to use based on input type.
-        # Reduce conversion cost if input is in Numpy
-        if isinstance(feature_columns, dict):
-            features_tensor = {
-                key: convert_pandas_to_torch_tensor(
-                    data,
-                    feature_columns[key],
-                    dtypes[key] if isinstance(dtypes, dict) else dtypes,
-                    unsqueeze=unsqueeze,
+        dtype: Optional[Union[torch.dtype, Dict[str, torch.dtype]]] = None,
+    ) -> pd.DataFrame:
+        if len(data.columns) == 1:
+            column_name = data.columns[0]
+            if isinstance(dtype, dict):
+                dtype = dtype[column_name]
+            model_input = torch.from_numpy(
+                convert_pandas_to_batch_type(data, type=np.ndarray)
+            ).to(dtype=dtype)
+        else:
+            array_dict = convert_pandas_to_batch_type(data, type=dict)
+            model_input = {
+                k: torch.from_numpy(v).to(
+                    dtype=dtype[k] if isinstance(dtype, dict) else dtype
                 )
-                for key in feature_columns
+                for k, v in array_dict.items()
             }
-        else:
-            features_tensor = convert_pandas_to_torch_tensor(
-                data,
-                columns=feature_columns,
-                column_dtypes=dtypes,
-                unsqueeze=unsqueeze,
-            )
-        return features_tensor
-
-    def _predict(self, tensor: torch.Tensor) -> pd.DataFrame:
-        """Handle actual prediction."""
-        prediction = self.model(tensor).cpu().detach().numpy()
-        # If model has outputs a Numpy array (for example outputting logits),
-        # these cannot be used as values in a Pandas Dataframe.
-        # We have to convert the outermost dimension to a python list (but the values
-        # in the list can still be Numpy arrays).
-        return pd.DataFrame({"predictions": list(prediction)}, columns=["predictions"])
+
+        self.model.eval()
```

Comment:
do we need `with torch.no_grad()` here?

Reply:
Added!

Reply:
Seems like we want both torch.no_grad and model.eval https://stackoverflow.com/questions/55627780/evaluating-pytorch-models-with-torch-no-grad-vs-model-eval